### PR TITLE
Add more Promotion fields

### PR DESCRIPTION
**Pull request by filipe-amberflo-io**

### CODE REVIEW

File: promotions_client.go [hunk 1]
```diff
@@ -39,12 +39,51 @@ type RemovePromotionRequest struct {
...
+	CycleMaxDiscount             *float64                     `json:"cycleMaxDiscount,omitempty"`
+	DiscountMap                  *map[string]float64          `json:"discountMap,omitempty"`
+	Measure                      *Measure                     `json:"measure,omitempty"`
+	PriceToDiscountMap           *map[string]float64          `json:"priceToDiscountMap,omitempty"`
+	DiscountCalculationStrategy  *string                      `json:"discountCalculationStrategy,omitempty"`
+	PromotionTimeLimit           *HistoryRequirements         `json:"promotionTimeLimit,omitempty"`
+	AcrossBillingPeriods         *bool                        `json:"acrossBillingPeriods,omitempty"`
+	TargetPlanId                 *string                      `json:"targetPlanId,omitempty"`
+	// TODO condition *PromotionCondition
+	// TODO promotionModel *PromotionModel
```
Comment (dor-amberflo) [inline promotions_client.go:86]: These 2 are probably not that important. They only work with the generic promotions, which are not used AFAIK by any customer.

**Review by dor-amberflo — APPROVED** [review]
Seems ok to me. I have a couple of ideas on how we can sort of simplify it, in terms of what we return, but they just end up obscuring the structure of the promotion.
This definitely shows the weakness of our codebase. Funnily enough, it's probably a codebase that would work well with js or python since json manipulation is very free there, maybe even java is not that convenient.
LGTM

**Comment by dor-amberflo** [comment]
One comment that I'm not sure about is how omitempty would work for the consumer. wouldn't they stilll need to find what's there? how would they go about sifting through the promotion object.

**Review by talha-aftab — APPROVED** [review]

**Comment by filipe-amberflo-io** [comment]
> One comment that I'm not sure about is how omitempty would work for the consumer. wouldn't they stilll need to find what's there? how would they go about sifting through the promotion object.

I think in this particular case `omitempty` is unnecessary. From what I understand, it afects serialization, not deserialization.

**Pull request merged**